trucks.js

```diff
@@ -503,42 +505,118 @@ var foodTrucks = [
 	}
 ];
 
+// getTrucks() - return all trucks
 var _ = require('underscore');
 
 function getTrucks() {
 	return foodTrucks;
 }
 
+//getTruck(name) - return the truck object matching 'name'
 function getTruck(name) {
 
-	console.log('name: ', name);
-	var truck = _.find(foodTrucks, function (truck) {
-			console.log('truck.name: ', truck.name);
-			return truck.name === name;
-		});
+	for (var i = 0, l = foodTrucks.length; i < l; i++) {
+
+		// Loop through food trucks until you find the matching name. When you find the name, stop.
+		if (foodTrucks[i].name.toLowerCase() === name.toLowerCase()) {
+
+			// return entire object for truck
+			return foodTrucks[i];
+		}
+	}
+
+	return false;
+}
+
+
+// getFoodTypes() - return unique list of all associated food types (underscore has a function to help)
+function getFoodTypes() {
+
+	// Returns list of food types for all trucks
+	var allFoodTypes = _.uniq(_.flatten(_.pluck(foodTrucks, 'type')));
 
-	console.log('truck: ', truck);
-	return truck;
+	allFoodTypes = _.sortBy(allFoodTypes, function (i) {
+		return i.toLowerCase();
+	});
+
+	return allFoodTypes;
 }
 
-function addTruck(truck) {
-	console.log('addTruck: ', truck);
-	if (truck) {
-		foodTrucks.push(truck);
+
+// filterByFoodType(foodType) - return trucks with associated 'foodType'
+function filterByFoodType(type) {
+
+	// Filter returns the type of food that match type
+	var matchingFoodTypes = _.filter(foodTrucks, function (item) {
+
+		if (typeof item.type === 'undefined') {
+			return false;
+		} else {
+			if (item.type.indexOf(type) === -1) {
+				return false;
+			} else {
+				return true;
+			}
+		}
+	});
+
+	return matchingFoodTypes;
+}
+
+
+// filterByDay(day) - return trucks with 'day' in schedule (use your filterByDay function from Module 3 homework)
+function filterByDay(day) {
+
+	// Filter returns the subset of foodTrucks that match 'day'
+	var sortedTrucks = _.filter(foodTrucks, function (item) {
+
+		// Exclude truck if object doesn't include 'schedule' prop, otherwise check for 'day' match
+		if (typeof item.schedule === 'undefined') {
+			return false;
+		} else {
+			return (item.schedule.indexOf(day) !== -1);
+		}
+	});
+
+	return sortedTrucks;
+}
+
+// Add truck
+function addTruck(name) {
+
+	var addNewTruck = [];
+
+	for (var i = 0; i < foodTrucks.length; i++) {
+		if (foodTrucks[i] === foodTrucks[i].name) {
+			addNewTruck.push(i);
```

Comment:
This is just adding numbers to the addNewTruck array, which is gone once the function returns.